tools/report_generators/cli.py

```diff
@@ -0,0 +1,291 @@
+        else:
+            raise FileNotFoundError(
+                f"Directory {wwwroot} missing. "
+                "To put the html into github repo for webhosting, run "
+                "`git clone git@github.com:oscovida/oscovida.github.io.git"
+                "wwwroot` or similar"
+            )
+
+
+def get_country_list():
+    d = oscovida.fetch_deaths()
+    c = oscovida.fetch_cases()
+
+    countries = d.index
+    countries2 = c.index
+    assert (countries2 == countries).all()
+
+    # Here we should identify regions in countries, and process those.
+    # Instead, as a quick hack to get started, we'll just take one country
+    # and the current "get_country" method will sum over all regions of one country if only
+    # the country name is given.
+
+    return sorted(countries.drop_duplicates())
+
+
+def generate_reports_countries(*, workers, kernel_name, wwwroot, force, disable_pbar, debug):
+    d = oscovida.fetch_deaths()
+    c = oscovida.fetch_cases()
+
+    countries = d.index
+    countries2 = c.index
+    assert (countries2 == countries).all()
+
+    #  TODO: The get_x_list methods should be part of Reporter class
+    countries = get_country_list()
+
+    cre = ReportExecutor(
+        Reporter=CountryReport,
+        wwwroot=wwwroot,
+        expiry_hours=2,
+        attempts=3,
+        workers=workers,
+        force=force,
+        disable_pbar=disable_pbar,
+        debug=debug,
+    )
+
+    if debug:
+        countries = countries[:10]
+
+    cre.create_html_reports(countries)
+
+    cre.create_markdown_index_page()
+
+
+def get_germany_regions_list():
+    data_germany = oscovida.fetch_data_germany()
+    land_kreis = data_germany[["Bundesland", "Landkreis"]]
+    ordered = land_kreis.sort_values(["Bundesland", "Landkreis"])
+    return ordered.drop_duplicates().values.tolist()
+
+
+def generate_reports_germany(*, workers, kernel_name, wwwroot, force, disable_pbar, debug):
+    _ = oscovida.fetch_data_germany()
+
+    #  TODO: The get_x_list methods should be part of Reporter class
+    germany_regions = get_germany_regions_list()
+
+    # data cleaning: on 13 April, we had a Landkreis "LK Göttingen (alt)"
+    # with only one data point. This causes plots to fail, because there
+    # is nothing to plot, and then the legend() command failed.
+    # We assume that the RKI labels unusual data with '(alt)', and remove those.
+
+    alt_data_sets = ["(alt)" in r[1].lower() for r in germany_regions]
+    if any(alt_data_sets):
+        bad_datasets = list(compress(germany_regions, alt_data_sets))
+
+        print(f"Removing datasets label with '(alt)': {bad_datasets}")
+
+        for bd in bad_datasets:
+            c, d, _ = oscovida.germany_get_region(landkreis=bd[1])
+            print(f"\tremoved: {bd} : len(cases)={len(c)}, len(deaths)={len(d)}")
+
+        bad_indices = list(compress(range(len(alt_data_sets)), alt_data_sets))
+
+        [germany_regions.pop(i) for i in bad_indices]
```

Comment:
unnecessary creation of list through comprehension here. Just use simple for loop.